resources-service/src/resource/resource-model.ts

```diff
@@ -82,52 +86,100 @@ export type ReferrableResource = {
+    searchResourcesByName: async (
+      accountSid: AccountSID,
+      {
+        nameSubstring,
+        ids = [],
+        pagination: { limit: unboundedLimit, start },
+      }: SimpleSearchParameters,
+    ): Promise<{ totalCount: number; results: ReferrableResourceSearchResult[] }> => {
+      const limit = Math.min(MAX_SEARCH_RESULTS, unboundedLimit);
+      const { results: idsOfNameMatches, totalCount: nameSearchTotalCount } = nameSubstring
+        ? await getWhereNameContains(accountSid, nameSubstring, start, limit)
+        : EMPTY_RESULT;
+
+      // If I'd known this logic would be such a hairball when I started writing the tests I wouldn't have bothered
+      // Still, it is only temporary, once the real search is implemented we won't need to splice 2 sets of results together so it will be much simpler
+      const idsToLoad = [...idsOfNameMatches, ...ids.filter(id => !idsOfNameMatches.includes(id))];
+      if (!idsToLoad.length) return { results: [], totalCount: nameSearchTotalCount };
+      // This might well be more than needed to meet the 'limit' criteria but best to query them all in case any of them 'miss'
+      const unsortedResourceList = await getByIdList(accountSid, idsToLoad);
+      const resourceMap = Object.fromEntries(
+        unsortedResourceList.map(resource => [resource.id, resource]),
+      );
+
+      // Add ALL the resources found looking up specific IDs to the paginated block of name search results
+      const untrimmedResults = idsToLoad
+        .map(id => {
+          const mappedValue = resourceMap[id];
+          // So each value is only used once
+          delete resourceMap[id];
+          return mappedValue;
+        })
+        .filter(r => r);
+
+      // Figure out the proper results for the specified pagination window from the above combined set
+      const resultsStartIndex = Math.max(0, start - nameSearchTotalCount); // If the start point is past the end of those returned in the name search, we need to drop some from the start of the result set to return the correct paginated window
+      const totalCount = nameSearchTotalCount + (untrimmedResults.length - idsOfNameMatches.length);
+      const results = untrimmedResults.slice(resultsStartIndex, resultsStartIndex + limit);
+      return {
+        results: results.map(record => ({
+          ...record,
+          attributes: attributeObjectGraphFromKeys(record.attributes),
+        })),
+        totalCount,
+      };
+    },
+
+    searchResources: async (
+      accountSid: AccountSID,
+      searchParameters: SearchParameters,
+    ): Promise<{ totalCount: number; results: ReferrableResourceSearchResult[] }> => {
+      const {
+        pagination: { limit: unboundedLimit, start },
+      } = searchParameters;
+      const limit = Math.min(MAX_SEARCH_RESULTS, unboundedLimit);
+      const { total, items } = await cloudSearchClient.search(
+        accountSid,
+        mapSearchParametersToKhpTermsAndFilters(searchParameters),
```

Comment:
Yes, that's correct, I don't know exactly how the mechanism for 'swapping out' different helpline specific mappings will work just now, but I imagine it will involve using different implementations of this mapping